### PR TITLE
Dev: Remove the `Security` and `Validation` ignore errors in phpstan-baseline.php

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
See #7731 

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Security/Exceptions/SecurityException.php [hunk 1]
```diff
@@ -16,20 +16,20 @@
 
 class SecurityException extends FrameworkException implements HTTPExceptionInterface
 {
-    public static function forDisallowedAction()
+    public static function forDisallowedAction(): SecurityException
```
Comment (paulbalandan) [inline system/Security/Exceptions/SecurityException.php:19]: as kenjis mentioned, please revert this and use `@return self` instead

**Review by paulbalandan — CHANGES_REQUESTED** [review]

File: system/Security/Exceptions/SecurityException.php [hunk 2]
```diff
@@ -16,20 +16,20 @@
 
 class SecurityException extends FrameworkException implements HTTPExceptionInterface
 {
-    public static function forDisallowedAction()
+    public static function forDisallowedAction(): self
```
Comment (kenjis) [inline system/Security/Exceptions/SecurityException.php:19]: This is a breaking change.
https://3v4l.org/GGS0f
Please use doc comment.

File: system/Validation/Exceptions/ValidationException.php [hunk 3]
```diff
@@ -15,27 +15,27 @@
 
 class ValidationException extends FrameworkException
 {
-    public static function forRuleNotFound(?string $rule = null)
+    public static function forRuleNotFound(?string $rule = null): ValidationException
```
Comment (kenjis) [inline system/Validation/Exceptions/ValidationException.php:18]: This is a breaking change.
Please use doc comment.

File: system/Security/Exceptions/SecurityException.php (hunk 2 above)
Comment (datamweb) [reply system/Security/Exceptions/SecurityException.php:19]: There are good explanations.
https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#breaking-changes

File: system/Validation/Exceptions/ValidationException.php (hunk 3 above)
Comment (ping-yee) [reply system/Validation/Exceptions/ValidationException.php]: Done.

File: system/Security/Exceptions/SecurityException.php (hunk 2 above)
Comment (ping-yee) [reply system/Security/Exceptions/SecurityException.php]: Done.

File: system/Security/Exceptions/SecurityException.php (hunk 1 above)
Comment (ping-yee) [reply system/Security/Exceptions/SecurityException.php]: Done.

**Review by kenjis — APPROVED** [review]

File: system/Security/Exceptions/SecurityException.php [hunk 4]
```diff
@@ -16,11 +16,24 @@
 
 class SecurityException extends FrameworkException implements HTTPExceptionInterface
 {
+    /**
+     * Throws when some specific is not allowed.
```
Comment (paulbalandan) [inline system/Security/Exceptions/SecurityException.php:20]:
```suggestion
     * Throws when some specific action is not allowed.
```

File: system/Validation/Exceptions/ValidationException.php [hunk 5]
```diff
@@ -15,26 +15,51 @@
...
+     *
+     * @return static
+     */
     public static function forRuleNotFound(?string $rule = null)
     {
         return new static(lang('Validation.ruleNotFound', [$rule]));
     }
 
+    /**
+     * Throws when the group value of config is not isset.
```
Comment (paulbalandan) [inline system/Validation/Exceptions/ValidationException.php:29]:
```suggestion
     * Throws when the group value of config is not set.
```

File: system/Security/Exceptions/SecurityException.php (hunk 4 above)
Comment (ping-yee) [reply system/Security/Exceptions/SecurityException.php]: Done.

File: system/Validation/Exceptions/ValidationException.php (hunk 5 above)
Comment (ping-yee) [reply system/Validation/Exceptions/ValidationException.php]: Done.

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**